### PR TITLE
Prevent Dragon Body from Despawning

### PR DESCRIPTION
**Pull request by Cojoke-dot**
## About the PR

Dragon is now unrevivable and the Dragons dead body will no longer despawn after a period
## Why / Balance

#29836, make Dragon unrevivable because there is nobody that should be reviving a dragon...
## Technical details

`RiftAccumulator` now only counts up when `RiftAccumulatorActive` is true, which only becomes false when the dragon dies
## Media

(this is not up to date but I'm too lazy to fix it, you can revive the dragon)

https://github.com/space-wizards/space-station-14/assets/83733158/057d433a-80d9-42d6-a1ec-336b35b49acb

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- fix: Dead Space Dragons no long despawn

### CODE REVIEW

**Comment by Cojoke-dot** [comment]
That feels like it's just going to end badly for everyone still. If someone somehow got a dragon body(which would be full of crew bodies) and revived it, the dragon would just end up killing said person or dying quickly due to low health. The main purpose of the PR is to prevent dragons already killed from despawning with all the bodies they ate. If someone defibs the dragon into a crit state and it vanishes while they are healing it, it's not going to be fun for the antag that spent their time trying to revive them or all the people that got eaten by the dragon.

**Comment by deltanedas** [comment]
being revived should reset the timer or remove the rift part, so the revived dragon is a safe pet

File: Content.Server/Dragon/DragonSystem.cs [hunk 1]
```diff
@@ -92,7 +92,8 @@ public override void Update(float frameTime)
                 }
             }
 
-            comp.RiftAccumulator += frameTime;
```
Comment (metalgearsloth) [inline Content.Server/Dragon/DragonSystem.cs:95]: Isn't it just easier to just... not accumulate these if the dragon is dead?

File: Content.Server/Dragon/Components/DragonComponent.cs [hunk 2]
```diff
@@ -32,6 +32,9 @@ public sealed partial class DragonComponent : Component
         [ViewVariables(VVAccess.ReadWrite), DataField("riftAccumulator")]
         public float RiftAccumulator = 0f;
 
+        [ViewVariables(VVAccess.ReadWrite), DataField("riftAccumulatorActive")]
+        public bool RiftAccumulatorActive = true;
+
```
Comment (metalgearsloth) [inline Content.Server/Dragon/Components/DragonComponent.cs:37]: Please update this to use modern standards as VVRW is unnecessary and datafield strings are obsolete though realistically this needs removing.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Comment by Brandon-Huu** [comment]
> make Dragon unrevivable because there is nobody that should be reviving a dragon...
Carp suit

**Comment by Cojoke-dot** [comment]
> Carp suit

Carp suit just gives you an AI team, player-driven carp and dragon are still going to eat you alive.

**Comment by Cojoke-dot** [comment]
> being revived should reset the timer or remove the rift part, so the revived dragon is a safe pet

They are still full of bodies… the main purpose of the pr is to make it so you don't lose those players due to a bug. Maybe in another pr though, there would need to be a system to get the people out without destroying the dragon body.

**Comment by Errant-4** [comment]
Okay hold on, a though just occured to me. Is it possible to harvest ichor from a dead dragon? If yes, is there a limit to the amount you can get? Or did we just create Medical's newest resource asset

**Comment by Cojoke-dot** [comment]
> Okay hold on, a though just occured to me. Is it possible to harvest ichor from a dead dragon? If yes, is there a limit to the amount you can get? Or did we just create Medical's newest resource asset

No, so how Ichor is created is that it's just added when a dragon eats someone. The issue with harvesting it is that a dragon metabolizes at an insanely high rate, do it only ends up in their blood for about 5 seconds.

**Comment by deltanedas** [comment]
> Carp suit just gives you an AI team, player-driven carp and dragon are still going to eat you alive.

roleplaying game when people roleplay:

**Comment by HelpTheSteak** [comment]
You could also just alter the code for pacifist to player carps and dragons but only make it apply to carp suit wearers (I don't know if this is possible tbh)

**Comment by Cojoke-dot** [comment]
> You could also just alter the code for pacifist to player carps and dragons but only make it apply to carp suit wearers (I don't know if this is possible tbh)

I'm going to do that anyways so the dragon and carps don't FF so much, but not in this pr 🙃

**Comment by IProduceWidgets** [comment]
Does unrevivable prevent it becoming a zombie? Because zombie dragon is pretty peak.

**Comment by Cojoke-dot** [comment]
> Does unrevivable prevent it becoming a zombie? Because zombie dragon is pretty peak.

I don't think so, humans don't become zombies when unrevivable. I'm not in a place to test it right now.

**Comment by IProduceWidgets** [comment]
I can just add the comp and vv and check. One moment.

**Comment by deltanedas** [comment]
> I'm going to do that anyways so the dragon and carps don't FF so much, but not in this pr 🙃

zombies have code for preventing ff for zombies, could be change to work with factions then have zombies use zombie faction and this use dragon faction

**Review by Emisse — APPROVED** [review]

**Comment by metalgearsloth** [comment]
Dragon code stinky but this is fine I suppose.

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**